### DISCUSSION TITLE
Watchtower not updating HomeAssistant

### DISCUSSION THREAD

**Discussion by coderunway** [containrrr/watchtower#1049] [Q&A]
I have setup watchtower to auto-update all my containers. It works perfectly for all but homeassistant container.  I see the following error in the logs while it's updating homeassistant. 

```
level=info msg="Unable to update container \"/homeassistant\": exposed ports does not match port bindings. Proceeding to next."
```

**exposed ports does not match port bindings**

My HA docker instance is running with ports 8123:8123 

Appreciate any help.

**Comment by piksel** [comment]
Is this on Synology? That is a check we added to avoid crashing on Synology devices since their version of Docker sometimes doesn't return the full container configuration. We currently don't know why this is, or how to correctly fill in the missing data.

**Reply by coderunway** [reply]
Yes, You are spot-on , it's a Synology. Ok that explains, it's working as designed, and is not a problem with my config :) Relief !! 

BTW, is this check specific to Home Assistant ?

**Reply by piksel** [reply]
No, we don't know why docker on Synology devices returns a partial configuration for some containers. There is probably a pattern, but we have not found one yet.

**Comment by tammert** [comment]
I've tinkered with my Synology a bit and it seems that some fields we are expecting aren't filled (at least not for `homeassistant` on the Synology device), so I've created a PR that should avoid the error: #1068
